### ISSUE TITLE
How to finetune grover?

### ISSUE THREAD

**Issue by chrislybaer** [gmum/huggingmolecules#24]
Do I see this correctly: your groover.forward returns the logits of the node-view fully connected neural network $p_{i, \text{node-view}}$ and the logits of the edge-view fully connected neural network $p_{i, \text{edge-view}}$ (appendix A.1 in https://arxiv.org/pdf/2007.02835.pdf) in
https://github.com/gmum/huggingmolecules/blob/main/src/huggingmolecules/models/models_grover.py#L118 

Thus, to finetune groover one trains on (for a single example i) $L(p_{i, \text{edge-view}}, y_i) + L(p_{i, \text{node-view}}, y_i) + \|p_{i, \text{edge-view}} - p_{i, \text{node-view}}\|_2 ?

Then to make predictions one uses $\sigma(p_{i, \text{edge-view}}) + \sigma(p_{i, \text{node-view}})?

**Comment by chrislybaer** [comment]
Also, it is unclear to me how to adapt: https://github.com/gmum/huggingmolecules/blob/main/experiments/src/training/training_lightning_module.py#L13 to do so.

**Comment by panpiort8** [comment]
Hi :) you are right - the objective for GROVER looks like $L(p_{i, \text{edge-view}}, y_i) + L(p_{i, \text{node-view}}, y_i) + |p_{i, \text{edge-view}} - p_{i, \text{node-view}}|_2$. It's implementation can be found [here](https://github.com/gmum/huggingmolecules/blob/main/experiments/src/training/training_loss_fn.py). However, the prediction does not use $\sigma$ function: it's simply the average: $p_{i, \text{edge-view}} + p_{i, \text{node-view}}. 

You actually don't need to adapt pytorch lightning module. The proper objective ([GroverLoss](https://github.com/gmum/huggingmolecules/blob/main/experiments/src/training/training_loss_fn.py)) is injected to the pytorch lightning module with GROVER's [configuration file](https://github.com/gmum/huggingmolecules/blob/df61a3b2f28a99ca42b02844abd5510aa99af263/experiments/configs/models/grover.gin#L6).

To fine-tune GROVER, simply follow [this section](https://github.com/gmum/huggingmolecules#running-scripts) of README.md.
